snowdialect.py

```diff
@@ -572,18 +572,49 @@ def get_schema_names(self, connection, **kw):
 
         return [self.normalize_name(row[1]) for row in cursor]
 
-    def get_table_comment(self, connection, table_name, schema=None, **kw):
+    def _get_table_comment(self, connection, table_name, schema=None, **kw):
         """
-        Returns comment of table in a dictionary as described by SQLAlchemy spec
+        Returns comment of table in a dictionary as described by SQLAlchemy spec.
         """
-        sql_command = "SHOW /* sqlalchemy:get_table_comment */ " \
+        sql_command = "SHOW /* sqlalchemy:_get_table_comment */ " \
                       "TABLES LIKE '{0}'{1}".format(
                                         table_name,
                                         (' IN SCHEMA {0}'.format(self.normalize_name(schema))) if schema else ''
                                     )
         cursor = connection.execute(sql_command)
-        ans = cursor.fetchone()
-        return {'text': ans['comment'] if ans['comment'] else None}
+        result = cursor.fetchone()
+        return result['comment']
+
+    def _get_view_comment(self, connection, table_name, schema=None, **kw):
+        """
+        Returns comment of view in a dictionary as described by SQLAlchemy spec.
+        """
+        sql_command = "SHOW /* sqlalchemy:_get_view_comment */ " \
+            "VIEWS LIKE '{0}'{1}".format(
+                table_name,
+                (' IN SCHEMA {0}'.format(self.normalize_name(schema))) if schema else ''
+            )
+        cursor = connection.execute(sql_command)
+        result = cursor.fetchone()
+        return result['comment']
+
+    def get_table_comment(self, connection, table_name, schema=None, **kw):
+        """
+        Returns comment associated with a table (or view) in a dictionary as
+        SQLAlchemy expects. Note that since SQLAlchemy may not (in fact,
+        typically does not) know if this is a table or a view, we have to
+        handle both cases here.
+        """
+        try:
+            comment = self._get_table_comment()
+            if comment is None or comment == "":
+                # the "table" being reflected is actually a view
+                comment = self._get_view_comment()
+        except (KeyError, TypeError):
+            # Also, we should not throw an error here just because we could not
+            # fetch the *comment* field
+            comment = None
```

Comment:
@keller00 I've found that calling e.g. `result['comment']` on an empty `result` row throws an error that SQLAlchemy _doesn't catch_, so it actually just breaks reflection -- hence wanting to guard against errors here and just return `None`.

Reply:
You're right, but then this whole PR won't work.
When trying to retrieve a VIew comment it will throw an exception in `_get_table_comment` and then this will catch it and not try to retrieve the view comment

Reply:
In light of this, maybe `return result` in the internal functions instead and check in this outer function `if result is None:`.
But I'd still remove this error catching.